src/main/java/io/jenkins/plugins/sample/OnboardingSectionConfiguration.java

```diff
@@ -0,0 +1,245 @@
+package io.jenkins.plugins.sample;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+import org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.*;
+import java.util.regex.Pattern;
+
+@Extension
+public class OnboardingSectionConfiguration extends GlobalConfiguration {
+
+    private String name;
+
+    private String description;
+
+    private String url;
+
+    private String username;
+
+    private String password;
+
+    private CategoryConfig categoryConfig;
+
+    public OnboardingSectionConfiguration() {
+        load();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
```

Comment:
FWIW I think it is typical to use `@DataBoundSetter` for `GlobalConfiguration` setters and `@DataBoundConstructor` for the constructor, though perhaps it is not required? See for example https://github.com/jenkinsci/configuration-as-code-plugin/blob/8b9efc1b9026942064d5248c92bf89cb00f83ea0/plugin/src/main/java/io/jenkins/plugins/casc/CasCGlobalConfig.java#L38.